### PR TITLE
Rewrite analysis tasks

### PR DESCRIPTION
**Pull request by rhcarvalho**
Simplify code and tests moving away from factories and mocks.

This makes the analysis checks take structured data, becoming easy to test and implement.

Take API type definitions from OpenShift/Kubernetes, moving away from embedded anonymous structs which were notably hard to instantiate.

### CODE REVIEW

**Comment by philbrookes** [comment]
@rhcarvalho when I run this branch I still get all the error output, shouldn't stderr be much quieter now?

File: analysis.go [hunk 1]
```diff
@@ -36,222 +39,141 @@ type Info struct {
...
+type definitionLoader struct {
+	basepath, project string
+	err               error
+}
+
+func (l *definitionLoader) Load(kind string, v interface{}) {
+	if l.err != nil {
+		return
+	}
+	path := filepath.Join(l.basepath, "projects", l.project, "definitions", kind+".json")
```
Comment (philbrookes) [inline analysis.go]: Is there a reason this only loads definitions, and not logs, or logs-previous, or nagios, etc?

Could it instead be a `projectEntityLoader` with `LoadDefinition`, `loadLog`, etc methods?

**Comment by rhcarvalho** [comment]
> @rhcarvalho when I run this branch I still get all the error output, shouldn't stderr be much quieter now?

That's because it is not rebase on top of latest master, with the log changes in.

If you pull `refs/pull/52/merge`, it should be quieter.

I can rebase and push force as well if you prefer.

**Comment by philbrookes** [comment]
> That's because it is not rebase on top of latest master, with the log changes in.

Does it make sense to do that before merging into master?

**Comment by rhcarvalho** [comment]
> > That's because it is not rebase on top of latest master, with the log changes in.
> 
> Does it make sense to do that before merging into master?

Travis runs tests against the merged branch, not the PR head:

https://travis-ci.org/feedhenry/fh-system-dump-tool/builds/159418417#L120

That's what I would expect from CI anyway :)

We only need a manual rebase if there are conflicts to be resolved. Otherwise every contributor would need to be constantly rebasing while waiting in the merge queue of a busy repository...

File: analysis.go (hunk 1 above)
Comment (rhcarvalho) [inline analysis.go]: One advantage of being a `definitionLoader` is that we can talk in terms of the `DefinitionLoader` interface, which could really be called `Loader` if we were to have specific loaders for other things.

What bothers me more here is that we have the knowledge about where definitions are stored in two different places: here and where they are written (by the DumpRunner).

**Comment by philbrookes** [comment]
> Travis runs tests against the merged branch, not the PR head

Awesome!

**Comment by philbrookes** [comment]
:+1: LGTM

**Pull request merged**